salt/utils/cache.py

```diff
@@ -130,41 +167,50 @@ def __delitem__(self, key):
         del self._key_cache_time[key]
         self._dict.__delitem__(key)
         # Do the same as the parent but also persist
-        self._write()
+        self.store()
 
     def _read(self):
         '''
         Read in from disk
         '''
-        if not HAS_MSGPACK or not os.path.exists(self._path):
-            return
-        with salt.utils.files.fopen(self._path, 'rb') as fp_:
-            cache = salt.utils.data.decode(msgpack.load(fp_, encoding=__salt_system_encoding__))
-        if "CacheDisk_cachetime" in cache:  # new format
-            self._dict = cache["CacheDisk_data"]
-            self._key_cache_time = cache["CacheDisk_cachetime"]
-        else:  # old format
-            self._dict = cache
-            timestamp = os.path.getmtime(self._path)
-            for key in self._dict:
-                self._key_cache_time[key] = timestamp
-        if log.isEnabledFor(logging.DEBUG):
-            log.debug('Disk cache retrieved: %s', cache)
-
-    def _write(self):
-        '''
-        Write out to disk
-        '''
-        if not HAS_MSGPACK:
-            return
-        # TODO Add check into preflight to ensure dir exists
-        # TODO Dir hashing?
-        with salt.utils.files.fopen(self._path, 'wb+') as fp_:
-            cache = {
-                "CacheDisk_data": self._dict,
-                "CacheDisk_cachetime": self._key_cache_time
-            }
-            msgpack.dump(cache, fp_, use_bin_type=True)
+        if msgpack is None:
+            log.error('Cache cannot be read from the disk: msgpack is missing')
+        elif not os.path.exists(self._path):
+            log.debug('Cache path does not exist for reading: %s', self._path)
+        else:
+            try:
+                with salt.utils.files.fopen(self._path, 'rb') as fp_:
+                    cache = salt.utils.data.decode(msgpack.load(fp_, encoding=__salt_system_encoding__))
```

Comment:
This looks like it will stacktrace if msgpack is missing.

Reply:
Hmm, how? Line #176 will just log error about it and return `None`. Unit tests are also explicitly testing this [here](https://github.com/saltstack/salt/pull/50585/files#diff-9fc11a5086d51e0723e81f3cf065bf0aR171) and implicitly [here](https://github.com/saltstack/salt/pull/50585/files#diff-9fc11a5086d51e0723e81f3cf065bf0aR374) and [here](https://github.com/saltstack/salt/pull/50585/files#diff-9fc11a5086d51e0723e81f3cf065bf0aR362). Am I still missing something?

Reply:
I think this should be fine. We are only entering else if `msgpack` is not `None`.

Reply:
Wow, what was I thinking there? My fault. Apologies.